test/tbb/test_numa_dist.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+    Copyright (c) 2005-2021 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "common/test.h"
+
+#include <stdio.h>
+#include "tbb/parallel_for.h"
+#include "tbb/global_control.h"
+#include "tbb/enumerable_thread_specific.h"
+
+#include "common/config.h"
+#include "common/utils.h"
+#include "common/utils_concurrency_limit.h"
+#include "common/utils_report.h"
+#include "common/vector_types.h"
+#include "common/cpu_usertime.h"
+#include "common/spin_barrier.h"
+#include "common/exception_handling.h"
+#include "common/concepts_common.h"
+#include "test_partitioner.h"
+
+#include <cstddef>
+#include <vector>
+
+//! \file test_parallel_for.cpp
+//! \brief Test for [algorithms.parallel_for] specification
+
+#if _MSC_VER
+#pragma warning (push)
+// Suppress conditional expression is constant
+#pragma warning (disable: 4127)
+#if __TBB_MSVC_UNREACHABLE_CODE_IGNORED
+    // Suppress pointless "unreachable code" warning.
+    #pragma warning (disable: 4702)
+#endif
+#if defined(_Wp64)
+    // Workaround for overzealous compiler warnings in /Wp64 mode
+    #pragma warning (disable: 4267)
+#endif
+#define _SCL_SECURE_NO_WARNINGS
+#endif //#if _MSC_VER
+
+
+struct numa {
+    WORD processorGroupCount;
+    std::vector<DWORD> numaProcessors;
+    DWORD maxProcessors;
+    numa() : processorGroupCount(GetMaximumProcessorGroupCount()), maxProcessors(GetActiveProcessorCount(ALL_PROCESSOR_GROUPS)){   
+        numaProcessors.resize(processorGroupCount);
+       for (WORD i = 0; i < processorGroupCount; i++) {
+            this->numaProcessors[i] = GetActiveProcessorCount((i));
+        }        
+    }
+};
+
+
+void TestNumaDistribution(){
+	std::vector<DWORD> validateProcgrp;
+	validateProcgrp.resize(GetMaximumProcessorGroupCount());
+	PROCESSOR_NUMBER proc;
+	struct numa nodes;
+	GetThreadIdealProcessorEx(GetCurrentThread(), &proc);
+	int master_thread_proc_grp = proc.Group;
+	auto requested_parallelism = nodes.numaProcessors.at(master_thread_proc_grp);
```

Comment:
auto requested_parallelism = nodes.numaProcessors.at(master_thread_proc_grp) + additionalParallelism;
If additionalParallelism = 0 - we have a test case for threads = cores on the node
if additionalParallelism > 0 - we can check for overflow